### PR TITLE
[cinder-csi-plugin] Check volume path and volume ID before Expand

### LINKED ISSUE #1497: [cinder-csi-plugin] volume length is not checked and volumeID not verified during expand volume
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

https://github.com/kubernetes-csi/csi-test/blob/master/pkg/sanity/node.go#L697
https://github.com/kubernetes-csi/csi-test/blob/master/pkg/sanity/node.go#L716
failed (we are not using master branch now) 

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager(or other related binary) version:
- OpenStack version:
- Others:

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue this PR fixes(if applicable)**:
fixes #1497 

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/csi/cinder/nodeserver.go [hunk 1]
```diff
@@ -526,6 +526,17 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
 	}
 	volumePath := req.GetVolumePath()
+	if len(volumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
```
Comment (Fedosin) [inline pkg/csi/cinder/nodeserver.go:530]: "Volume path not provided"

File: pkg/csi/cinder/nodeserver.go [hunk 2]
```diff
@@ -526,6 +526,17 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
...
+	}
+
+	_, err := ns.Cloud.GetVolume(volumeID)
+	if err != nil {
+		if cpoerrors.IsNotFound(err) {
+			return nil, status.Error(codes.NotFound, "Volume not found")
```
Comment (Fedosin) [inline pkg/csi/cinder/nodeserver.go:536]: Maybe `"Volume with ID %s not found", volumeID`?

**Review by Fedosin — CHANGES_REQUESTED** [review]

File: pkg/csi/cinder/nodeserver.go [hunk 3]
```diff
@@ -526,6 +526,17 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
 	}
 	volumePath := req.GetVolumePath()
+	if len(volumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume path not provided")
+	}
+
+	_, err := ns.Cloud.GetVolume(volumeID)
```
Comment (ramineni) [inline pkg/csi/cinder/nodeserver.go:533]: @jichenjc From the issue details this is not required. 
https://github.com/kubernetes-csi/csi-test/blob/master/pkg/sanity/node.go#L682 , this checks for volumeID , which we are already doing in L525. So ideally shouldnt fail. Could you check again why that test case is failing.

Comment (jichenjc) [reply pkg/csi/cinder/nodeserver.go:533]: my bad, actually the test case is https://github.com/kubernetes-csi/csi-test/blob/master/pkg/sanity/node.go#L716
I updated issue just now

Comment (ramineni) [reply pkg/csi/cinder/nodeserver.go:533]: @jichenjc got it. I think this is not caught by current test as we are using old version of csi-test . Could you update the csi-test to new version as well to make sure this part of code is tested in CI as well. Thanks.

**Comment by jichenjc** [comment]
> 
> 
> @jichenjc Thanks for the PR. code changes looks good. But please update the csi-test module to latest version , so that this code change can tested in CI as well.

https://github.com/kubernetes/cloud-provider-openstack/pull/1496
this is the PR that handling the update, I am not sure combine several things together will make thing way more complated
so this is a seperated PR

**Comment by ramineni** [comment]
> > @jichenjc Thanks for the PR. code changes looks good. But please update the csi-test module to latest version , so that this code change can tested in CI as well.
> 
> #1496
> this is the PR that handling the update, I am not sure combine several things together will make thing way more complated
> so this is a seperated PR

@jichenjc I think combining should be fine , as we can actually see if test is fixed or any more changes required.

**Comment by jichenjc** [comment]
> 
> 
> > > @jichenjc Thanks for the PR. code changes looks good. But please update the csi-test module to latest version , so that this code change can tested in CI as well.
> > 
> > 
> > #1496
> > this is the PR that handling the update, I am not sure combine several things together will make thing way more complated
> > so this is a seperated PR
> 
> @jichenjc I think combining should be fine , as we can actually see if test is fixed or any more changes required. As I was wondering before why it wasnt caught on CI as sanity test is there

I indicated why it not failed at beginning in the issue :)

as mentioned, there might be more issues (actually I am working on) and it might cause more time 
https://github.com/kubernetes/cloud-provider-openstack/issues/1498

so if we can fix this now ,I think about how about let's merge it (anyway it's a bug...)

**Comment by jichenjc** [comment]
I find a way to fix #1498
so guess then we can combine those whole patch set

**Pull request closed without merging**